shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1478,20 +1830,63 @@ - (void)updateEditingStateWithDelta:(FlutterTextEditingDelta*)delta {
     @"deltas" : @[ deltaToFramework ],
   };
 
-  [self.textInputDelegate updateEditingClient:_textInputClient withDelta:deltas];
+  [self.textInputDelegate flutterTextInputView:self
+                           updateEditingClient:_textInputClient
+                                     withDelta:deltas];
 }
 
 - (BOOL)hasText {
   return self.text.length > 0;
 }
 
 - (void)insertText:(NSString*)text {
+  NSMutableArray<FlutterTextSelectionRect*>* copiedRects =
+      [[NSMutableArray alloc] initWithCapacity:[_selectionRects count]];
+  NSAssert([_selectedTextRange.start isKindOfClass:[FlutterTextPosition class]],
+           @"Expected a FlutterTextPosition for position (got %@).",
+           [_selectedTextRange.start class]);
+  NSUInteger insertPosition = ((FlutterTextPosition*)_selectedTextRange.start).index;
+  for (NSUInteger i = 0; i < [_selectionRects count]; i++) {
+    NSUInteger rectPosition = _selectionRects[i].position;
+    if (rectPosition == insertPosition) {
+      for (NSUInteger j = 0; j <= text.length; j++) {
+        [copiedRects
+            addObject:[FlutterTextSelectionRect selectionRectWithRect:_selectionRects[i].rect
+                                                             position:rectPosition + j]];
+      }
+    } else {
+      if (rectPosition > insertPosition) {
+        rectPosition = rectPosition + text.length;
+      }
+      [copiedRects addObject:[FlutterTextSelectionRect selectionRectWithRect:_selectionRects[i].rect
+                                                                    position:rectPosition]];
+    }
+  }
+
+  _scribbleFocusStatus = FlutterScribbleFocusStatusUnfocused;
+  [self resetScribbleInteractionStatusIfEnding];
+  _selectionRects = copiedRects;
```

Comment:
This is leaking the old value of `_selectionRects`, use `self.selectionRects = copiedRects; [copiedRects release];`